hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java

```diff
@@ -230,6 +242,11 @@ public void set(K id, P property, Object value) {
 			changedKeys.add(id);
 			return changeFactory.apply(id);
 		}), property, value);
+
+		final var merkleProperties = changes.get(id);
+		final var merkleLeaf = entities.contains(id) ? entities.getRef(id) : null;
```

Comment:
⚠️ This is risky! For example, [the `BackingAccounts.getRef()` method](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ledger/backing/BackingAccounts.java#L55) calls `getForModify()`, so by calling `getRef()` here we have opened the door to accidentally mutating a `MerkleAccount` even without `commit()` is called. (Hopefully there is no code path that would accidentally do such a mutation, but it's still an unnecessary risk. 😁)

We should return to the pattern you used [here](https://github.com/hashgraph/hedera-services/blob/b0d066b4606a6d28bd17735691f9a694bcad0f02/hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java#L197), where you only call `getRef()` within `commit()`, as you build the list of pending changes up front.